### PR TITLE
Update ScaFi and adjust/improve incarnation

### PR DESCRIPTION
**Pull request by metaphori**
- Update ScaFi version
- Adjust sensors in ScaFi incarnation 
- Implement other fixes and improvements

### CODE REVIEW

File: versions.properties [hunk 1]
```diff
@@ -164,3 +164,5 @@ version.net.sourceforge.pmd..pmd-java=version.net.sourceforge.pmd..pmd-core
 version.net.sourceforge.pmd..pmd-scala=version.net.sourceforge.pmd..pmd-core
 
 version.net.sourceforge.pmd..pmd-kotlin=version.net.sourceforge.pmd..pmd-core
+
+version.it.unibo.scafi..scafi-core_2.13=0.3.3
```
Comment (DanySK) [inline versions.properties:168]: There should also be one line deleted with the old scafi name

**Review by DanySK — CHANGES_REQUESTED** [review]

**Comment by DanySK** [comment]
There's one smell in `versions.properties`, plus it apparently does not compile. Is this a draft?

File: versions.properties (hunk 1 above)
Comment (metaphori) [reply versions.properties:166]: Right

**Comment by DanySK** [comment]
CI failed due to Kotlin/dokka#1599. Restarting.

**Review by DanySK — APPROVED** [review]
Sonarcloud issues look like false positives to me. Take a look for the sake of meticulousness @metaphori.
In intend to rebase this in as soon as the CI passes, if Dokka doesn't fail on hashcodes

**Comment by DanySK** [comment]
@metaphori `Execution failed for task ':alchemist-incarnation-scafi:test'.` See https://scans.gradle.com/s/plnm4kzyk62dy

File: alchemist-incarnation-scafi/src/test/scala/it/unibo/alchemist/scafi/test/TestInSimulator.scala [hunk 2]
```diff
@@ -68,7 +68,7 @@ class TestInSimulator[P <: Position[P]] extends AnyFunSuite with Matchers {
       if(node.getId==0) {
         inputMolecule shouldBe 77
         outputMolecule shouldBe 177
-        val p = getValue[Poin                                                                     t3D](MOL_POSITION)
```
Comment (DanySK) [inline alchemist-incarnation-scafi/src/test/scala/it/unibo/alchemist/scafi/test/TestInSimulator.scala]: what a mastake! :rofl:

**Pull request merged**